dashboard/app/controllers/data_docs_controller.rb

```diff
@@ -32,4 +32,34 @@ def show
       dataDocContent: @data_doc.content,
     }
   end
+
+  # GET /data_docs/:key/edit
+  def edit
+    @data_doc = DataDoc.find_by(key: params[:key])
```

Comment:
up to you but if all the routes are fetching the data doc by key, it might be worth moving this into a private method and using `before_action :set_data_doc`, like in [scripts_controller.rb](https://github.com/code-dot-org/code-dot-org/blob/29fbcf43da3eb1a80753d682372af98a408ef481/dashboard/app/controllers/scripts_controller.rb#L8).

Reply:
Good call. I tried using `load_resource id_param: :key` but had no luck ...

Reply:
Done! Thank you